client/src/components/Card/index.js

```diff
@@ -1,34 +1,55 @@
 import React from 'react';
 import { Card, CardTitle, CardBody, CardHeader, Progress } from 'reactstrap';
-import './Card.css';
+import './Card.scss';
 import { Link } from 'react-router-dom';
+class PetCard extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      src: null
+    };
+  }
 
-const PetCard = props => {
-  return (
-    <div className='col-md-4 card-item'>
-      <Link to={`/pets/${props.address}`}>
-        <Card>
-          <CardHeader className='card-header'>
-            <img
-              alt='...'
-              className='img-center img-fluid item-img '
-              src='https://hips.hearstapps.com/ghk.h-cdn.co/assets/17/30/pembroke-welsh-corgi.jpg'
-            />
-          </CardHeader>
-          <CardBody>
-            <CardTitle>Name: {props.name}</CardTitle>
-            <Progress animated color='danger' value={props.age}>
-              Age
-            </Progress>
-            <br />
-            <Progress animated color='success' value={props.size}>
-              Size
-            </Progress>
-          </CardBody>
-        </Card>
-      </Link>
-    </div>
-  );
-};
+  componentDidMount() {
+    if (this.props.type === '1') {
+      this.setState({
+        src: require('assets/img/penguin.png')
+      });
+    }
+    if (this.props.type === '2') {
+      this.setState({
+        src: require('assets/img/dragon.png')
+      });
+    }
+    if (this.props.type === '3') {
+      this.setState({
+        src: require('assets/img/cat.png')
+      });
+    }
+  }
+  render() {
+    return (
+      <div className='col-md-4 card-item'>
+        <Link to={`/pets/${this.props.address}`}>
+          <Card>
+            <CardHeader className='card-header'>
+              <img alt='...' className='img-center img-fluid item-img' src={this.state.src} />
```

Comment:
code quá rườm rà, e code theo kiểu này sẽ gọn hơn rất nhiều
```suggestion 
let imgSrc = {1: require('assets/img/penguin.png'), 2: require('assets/img/dragon.png'), 3: require('assets/img/cat.png')}
src={imgSrc[this.props.type]}
```

Reply:
viết lại component dạng function còn gọn nữa
```js
const PetCard = {address, type, age, size } => ();
```